### PR TITLE
Add Int32 and Float32 type testing

### PR DESCRIPTION
**Pull request by SBGoods**
Add test cases for `int32` and `float32` function types and `float32` precision testing.

### CODE REVIEW

File: internal/framework6provider/float64_precision_resource_test.go [hunk 1]
```diff
@@ -72,3 +75,161 @@ func TestSchemaResource_Float64Attribute_Precision_Plan(t *testing.T) {
...
+
+// This test documents edge case behavior with math.MaxFloat32
+// This number is unique in that it has an exact representation in both float32 and float64,
+// it is also an integer, which may affect how this value gets encoded/decoded with go-cty.
+// https://github.com/hashicorp/terraform-plugin-framework/issues/1017
+func TestSchemaResource_Float64Attribute_Precision_MaxFloat32(t *testing.T) {
+	resource.UnitTest(t, resource.TestCase{
+		TerraformVersionChecks: []tfversion.TerraformVersionCheck{
+			// go-cty v1.14.4 uses string msgpack encoding instead of float msgpack encoding for large whole numbers
+			// https://github.com/hashicorp/terraform/commit/cd252557e4200e031b942d9ff35c455bb30d858f
```
Comment (austinvalle) [inline internal/framework6provider/float64_precision_resource_test.go:87]: Nit: Might be a better breadcrumb to link the PR: https://github.com/hashicorp/terraform/pull/34756, which has some useful notes + links to the go-cty PR

Similar note for the other tests and `framework5provider` equivalents.

File: internal/framework6provider/float64_precision_resource_test.go [hunk 2]
```diff
@@ -72,3 +75,161 @@ func TestSchemaResource_Float64Attribute_Precision_Plan(t *testing.T) {
...
+			// This changes allows the math.MaxFloat32 value to succeed in planning but fail during apply.
+			// Terraform v1.9.0-alpha20240501 is the first Terraform version to use this updated encoding.
```
Comment (austinvalle) [inline internal/framework6provider/float64_precision_resource_test.go:89]: Nit: Probably only really care about the stable Terraform versions, can just reference `v1.9.0`

Similar note for the other tests and `framework5provider` equivalents.

File: internal/framework6provider/float64_precision_resource_test.go [hunk 3]
```diff
@@ -72,3 +75,161 @@ func TestSchemaResource_Float64Attribute_Precision_Plan(t *testing.T) {
 		},
 	})
 }
+
+// This test documents edge case behavior with math.MaxFloat32
+// This number is unique in that it has an exact representation in both float32 and float64,
+// it is also an integer, which may affect how this value gets encoded/decoded with go-cty.
+// https://github.com/hashicorp/terraform-plugin-framework/issues/1017
```
Comment (austinvalle) [inline internal/framework6provider/float64_precision_resource_test.go:82]: Great explanation! ❤️

File: internal/framework6provider/float64_precision_resource_test.go [hunk 4]
```diff
@@ -72,3 +75,161 @@ func TestSchemaResource_Float64Attribute_Precision_Plan(t *testing.T) {
...
+//
+// go-cty v1.14.4 uses string msgpack encoding instead of float msgpack encoding for large whole numbers
+// https://github.com/hashicorp/terraform/commit/cd252557e4200e031b942d9ff35c455bb30d858f
+// This changes allows the math.MaxFloat32 value to succeed in planning but fail during apply.
+// Terraform v1.9.0-alpha20240501 is the first Terraform version to use this updated encoding.
+func TestSchemaResource_Float64Attribute_Precision_MaxFloat32_TF1_9(t *testing.T) {
+	resource.UnitTest(t, resource.TestCase{
+		TerraformVersionChecks: []tfversion.TerraformVersionCheck{
+			tfversion.SkipBelow(version.Must(version.NewVersion("1.9.0-alpha20240501"))),
```
Comment (austinvalle) [inline internal/framework6provider/float64_precision_resource_test.go:172]: Nit: If we can, let's just use the stable version for this `v1.9.0`

Similar note for the `framework5provider` equivalent.

**Review by austinvalle — APPROVED** [review]
All the tests here look good, left a couple notes about comments and version choices.

One more place to consider adding smoke tests for `Float32Attribute` and `Int32Attribute` is in the basic schema tests:
- `internal/framework5provider/schema_resource.go`
- `internal/framework5provider/schema_resource_test.go`
- `internal/framework6provider/schema_resource.go`
- `internal/framework6provider/schema_resource_test.go`

Great job! 🚀
